src/lib/merkle_ledger/database.ml

```diff
@@ -294,7 +299,26 @@ end)
 
   let copy {kvdb; sdb} = {kvdb= Kvdb.copy kvdb; sdb= Sdb.copy sdb}
 
-  let remove_accounts_exn _ _ = failwith "TODO: Implement"
+  let remove_accounts_exn t keys =
+    let locations =
+      (* if we don't have a location for all keys, raise an exception *)
+      let rec loop keys accum =
+        match keys with
+        | [] -> accum (* no need to reverse *)
+        | key :: rest -> (
+          match Account_location.get t key with
+          | Ok loc -> loop rest (loc :: accum)
+          | Error err -> raise (Db_error.Db_exception err) )
+      in
+      loop keys []
+    in
+    (* N.B.: we're not using stack database here to make available newly-freed locations *)
+    List.iter keys ~f:(Account_location.delete t) ;
+    List.iter locations ~f:(fun loc -> delete_raw t loc) ;
+    (* recalculate hashes for each removed account *)
+    List.iter locations ~f:(fun loc ->
```

Comment:
we could recursively remove any intermediate nodes both of whose children are empty as well, but it's ok to not do that for now.